### ISSUE TITLE
False negative when matching an unpacked iterable argument to a non-existent positional parameter

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4140]
```python
def g(x: int):
    ...

def func(y: tuple[int]):
    g(1, *y) # This should generate an error but it doesn't
```

This bug originates from this discussion: https://github.com/microsoft/pyright/discussions/4134#discussioncomment-4058879

**Comment by erictraut** [comment]
The challenge here is that we don't want to generate an error in these cases:

```python
def func(y: tuple[int, ...]):
    g(1, *y)
```

or

```python
def func(*args):
    g(1, *args)
```

**Comment by mehdigmira** [comment]
> The challenge here is that we don't want to generate an error in these cases:
> 
> ```python
> def func(y: tuple[int, ...]):
>     g(1, *y)
> ```
> 
> or
> 
> ```python
> def func(*args):
>     g(1, *args)
> ```

Why wouldn't we want to generate an error in those cases ?

```python
def func(*args):
    g(*args)
```

I would understand that this (even if somewhat dangerous) would not trigger an error.
But once you've passed a first argument to function `g`, adding anything else is just bad code that at best will do nothing (if args is empty), and at worst would break at runtime.

**Comment by erictraut** [comment]
Generating an error in these situations creates too many false positives in legitimate code. I tried making the change and ran it over internal Microsoft code bases, and it produced a bunch of false positive errors.

Our general philosophy for pyright is that we prefer false negatives over false positives when forced to make a choice between the two. This is especially true of basic-mode diagnostic checks. We can be a bit more opinionated for strict-mode checks. This is a basic-mode check, so we need to be more conservative about generating errors.

**Comment by erictraut** [comment]
This is included in pyright 1.1.279, which I just published. It will also be included in a future version of pylance.
